tests/test_tasks.py

```diff
@@ -0,0 +1,57 @@
+"""Tests for the tasks of the Imperial Coldfront plugin."""
+
+import datetime
+
+import pytest
+
+from imperial_coldfront_plugin.tasks import send_expiration_notifications
+
+
+class TestSendExpirationNotificationsTask:
+    """Tests for the send_expiration_notifications task."""
+
+    @pytest.mark.parametrize("days", (1, 5, 30))
+    def test_notification_sent_before_expiry(
+        self, days, pi, pi_group_membership, mailoutbox
+    ):
+        """Test that a notification is sent for memberships expiring in the future."""
+        pi_group_membership.expiration = datetime.date.today() + datetime.timedelta(
+            days=days
+        )
+        pi_group_membership.save()
+
+        send_expiration_notifications()
+
+        email = mailoutbox[0]
+        email.subject == "HPC Access Expiration Alert"
+        email.to == [pi.email, pi_group_membership.group.owner.email]
```

Comment:
Surprised this is passing... `pi` and pi_group_membership.group.owner` should be the same user. Also the signature of `send_expiration_alert_email` expects the group member email to be first.

Reply:
You are right. It fails when I use `assert`